lib/jekyll/url.rb

```diff
@@ -86,7 +86,11 @@ def generate_url_from_hash(template)
 
     def generate_url_from_drop(template)
       template.gsub(%r!:([a-z_]+)!) do |match|
-        replacement = @placeholders.public_send(match.sub(":".freeze, "".freeze))
+        begin
+          replacement = @placeholders.public_send(match.sub(":".freeze, "".freeze))
+        rescue NoMethodError
```

Comment:
Instead of a `begin .. rescue`, what about checking for the key first, like

```ruby
template.gsub(%r!:([a-z_]+)!) do |match|
  key = match.sub(":".freeze, "".freeze)
  raise NoMethodError, "No URL template key #{key} exists!" unless @placeholders.key?(key)
  if @placeholders[key].nil?
    "".freeze
  else
    self.class.escape_path(@placeholders[key])
  end
end.gsub(%r!//!, "/".freeze)
```